datafusion/core/tests/sql/expr.rs

```diff
@@ -200,6 +200,41 @@ async fn binary_bitwise_shift() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_comparison_func_expressions() -> Result<()> {
+    test_expression!("greatest(1,2,3)", "3");
+    test_expression!("least(1,2,3)", "1");
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_comparison_func_array_scalar_expression() -> Result<()> {
```

Comment:
Can we please write this test using `sqllogictest` instead of a new rs test? I think you'll find it quite nice for sql tests and it is easier to maintain and extend. 

https://github.com/apache/arrow-datafusion/tree/main/datafusion/core/tests/sqllogictests#cookbook-adding-tests